imap_processing/cli.py

```diff
@@ -407,32 +408,35 @@ def upload_products(self, products: list[Path]) -> None:
             A list of file paths to upload to the SDC.
         """
         if self.upload_to_sdc:
-            # Validate that the files don't already exist
-            for filename in products:
-                file_path = ScienceFilePath(filename)
-                existing_file = imap_data_access.query(
-                    instrument=file_path.instrument,
-                    data_level=file_path.data_level,
-                    descriptor=file_path.descriptor,
-                    start_date=file_path.start_date,
-                    end_date=file_path.start_date,
-                    repointing=file_path.repointing,
-                    version=file_path.version,
-                    extension="cdf",
-                    table="science",
-                )
-                if existing_file:
-                    raise ProcessInstrument.ImapFileExistsError(
-                        f"File {filename} already exists in the IMAP SDC. "
-                        "No files were uploaded."
-                        f"Generated files: {products}."
-                    )
-
-            if len(products) == 0:
+            if not products:
                 logger.info("No files to upload.")
+                return
+
             for filename in products:
-                logger.info(f"Uploading file: {filename}")
-                imap_data_access.upload(filename)
+                file_obj = generate_imap_file_path(filename)
+
+                if isinstance(file_obj, SPICEFilePath):
+                    # upload SPICE files (DPS)
+                    logger.info(f"Uploading SPICE file: {filename}")
+                    try:
+                        imap_data_access.upload(filename)
+                    except ProcessInstrument.ImapFileExistsError as e:
+                        logger.error(f"Failed to upload SPICE file {filename}: {e}")
+                        # Continue to upload next file
+                        continue
+
+                elif isinstance(file_obj, ScienceFilePath):
+                    logger.info(f"Uploading science file: {filename}")
+                    try:
+                        imap_data_access.upload(filename)
+                    except ProcessInstrument.ImapFileExistsError as e:
+                        logger.error(f"Failed to upload science file {filename}: {e}")
+                        # Continue to upload next file
+                        continue
+                else:
+                    logger.info(
+                        f"File type for {filename} is not supported for upload."
+                    )
```

Comment:
Why not make this totally generic:
```suggestion
                file_obj = generate_imap_file_path(filename)
                logger.info(f"Uploading {type(file_obj).__name__} object")
                try:
                    imap_data_access.upload(filename)
                except ProcessInstrument.ImapFileExistsError as e:
                    logger.error(f"Failed to upload file {filename}: {e}")
                    continue
```

Is there any reason why we can't just let the CLI upload any of the `ScienceFilePath, AncillaryFilePath, SPICEFilePath, QuicklookFilePath, DependencyFilePath` filetypes?

Reply:
Like, I wouldn't expect this code to upload quicklooks or dependency files, but is it that big a deal if it does?

Reply:
You also might want to expand the exception class here to cover any kind of generic "file couldn't upload" error, not just the file existing one.

Reply:
since `post_processing ` is only used in processing, we won't upload quicklooks or dependency file. So consolidating sounds ok to me. Sounds like you and Greg are on same page about just having one try/except. will do that.

Reply:
I think that Menlo will want to upload ancillary files from their processing as well.
https://github.com/IMAP-Science-Operations-Center/imap-data-access/pull/266

I have thought about doing quicklooks here for speed as well since those are high priority again apparently :( So keeping it open wider is appealing to me for all those reasons.